src/Select.purs

```diff
@@ -267,27 +271,33 @@ component =
     -- Just the normal Halogen eval
     eval :: QueryF o item ~> ComponentDSL o item m
     eval = case _ of
+      Initialize a -> a <$ do
+        ref <- H.liftEffect $ Ref.new Nothing
+        modifyState_ _ { debounceRef = Just ref }
+
       Search str a -> a <$ do
         st <- getState
+        ref :: Maybe Debouncer <- H.liftEffect $ map join $ traverse Ref.read st.debounceRef
         modifyState_ _ { search = str }
         setVis On
 
-        case st.inputType, st.debouncer of
+        case st.inputType, ref of
           TextInput, Nothing -> unit <$ do
             var   <- H.liftAff AVar.empty
             fiber <- H.liftAff $ forkAff do
               delay st.debounceTime
```

Comment:
Should probably guard this, too, so we don't even go through any of these steps unless a debounce time has actually been set.